src/main/java/org/fenixedu/academicextensions/services/registrationhistory/RegistrationHistoryReport.java

```diff
@@ -347,6 +350,11 @@ public LocalDate getLastRegistrationStateDate() {
         return state == null ? null : state.getStateDate().toLocalDate();
     }
 
+    public DateTime getLastEnrolmentUpdateDateTime() {
+        return registration.getCurriculumLineLogsSet().stream().filter(log -> log instanceof EnrolmentLog).map(log -> log.getDateDateTime())
+                .max(DateTime::compareTo).orElse(null);
```

Comment:
since `DateTime` already implements `Comparable`, you can do `.max(Comparator.naturalOrder())`